narwhal/types/src/primary.rs

```diff
@@ -1183,6 +1231,306 @@ impl CertificateV1 {
     }
 }
 
+// Holds AggregateSignatureBytes but with the added layer to specify the
+// signatures verification state. This will be used to take advantage of the
+// certificate chain that is formed via the DAG by only verifying the
+// non-parents of the certificate chain when they are fetched from validators
+// during catchup.
+#[derive(Clone, Serialize, Deserialize, MallocSizeOf, Debug)]
+pub enum AggregateSignatureVerificationState {
+    // This state occurs when the cert was a parent of another fetched certificate
+    // that was verified directly, then this certificate is verified indirectly.
+    VerifiedIndirectly(AggregateSignatureBytes),
+    // This state occurs when a certificate was either created locally, received
+    // via brodacast, or fetched but was not the parent of another certifiate.
+    // Therefore this certificate had to be verified directly.
+    VerifiedDirectly(AggregateSignatureBytes),
```

Comment:
Can we only keep the `AggregateSignatureBytes` for `VerifiedDirectly` and `Unverified` states?